### PR TITLE
Always copy ndarray when it is copied to another device

### LINKED ISSUE #732: BUG: Variable.to_gpu doesn't work to move the other GPU
I understand ``Variable.to_gpu`` moves current variable to specified GPU destructively. But it isn't the case when current / target devices are both GPU?

```
c1 = chainer.Variable(cp.asarray(np.array([1, 2, 3])))
cuda.get_device(c1.data)
# <CUDA Device 0>

# NG, must be Device 2?
c1.to_gpu(2)
cuda.get_device(c1.data)
# <CUDA Device 0>
 
# OK
c2 = F.copy(c1, 2)
cuda.get_device(c2.data)
# <CUDA Device 2>
```

### PR DESCRIPTION
**Pull request by unnonouno**
fix #732

### CODE REVIEW

**Comment by delta2323** [comment]
Do we already check `Variable.to_gpu` when:
* source and destination is same GPU
* source is CPU and destination is GPU

File: chainer/cuda.py [hunk 1]
```diff
@@ -190,7 +190,12 @@ def to_gpu(array, device=None, stream=None):
     check_cuda_available()
     assert stream is None  # TODO(beam2d): FIX IT
     with get_device(device):
-        return cupy.asarray(array)
+        if isinstance(array, cupy.ndarray) and \
+           int(array.data.device) != cupy.cuda.device.get_device_id():
```
Comment (beam2d) [inline chainer/cuda.py]: You can use the result of get_device() above to specify the destination GPU id.

Comment (unnonouno) [inline chainer/cuda.py]: Do you mean `int(get_device(array)) != cupy.cuda.device.get_device_id()` ?

**Comment by unnonouno** [comment]
`TestVariable` has `test_to_gpu_from_gpu`  and `test_to_gpu`.

File: chainer/cuda.py (hunk 1 above)
Comment (okuta) [inline chainer/cuda.py]: You can use `cupy.cuda.get_device_id()`.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**